### DISCUSSION TITLE
Issue with NextAuth Bungie Provider Configuration

### DISCUSSION THREAD

**Discussion by M7ilan** [nextauthjs/next-auth#8918] [Help]
### Provider type

Bungie

### Environment

  System:
    OS: Windows 10 10.0.19045
    CPU: (12) x64 Intel(R) Core(TM) i5-10400F CPU @ 2.90GHz
    Memory: 8.72 GB / 15.85 GB
  Binaries:
    Node: 18.12.1 - C:\Program Files\nodejs\node.EXE       
    npm: 8.19.2 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (117.0.2045.47)
    Internet Explorer: 11.0.19041.1566

### Reproduction URL

https://github.com/M7ilan/next-auth-bungie-provider

### Describe the issue

I'm facing an issue while configuring the NextAuth Bungie Provider. The error message I'm encountering is as follows:

```bash
Argument of type '{ clientId: string | undefined; clientSecret: string | undefined; headers: { "X-API-Key": string | undefined; }; }' is not assignable to parameter of type 'Partial<OAuthConfig<any>>'.
  Object literal may only specify known properties, and 'headers' does not exist in type 'Partial<OAuthConfig<any>>'.
```

I have attempted to set up the Bungie Provider within my NextAuth configuration, but I'm encountering this TypeScript error. I'm unsure why this error is occurring and how to resolve it. Below is the code snippet where the error is originating:

```tsx
// app\api\auth\[...nextauth]\route.ts
import NextAuth from "next-auth";
import BungieProvider from "next-auth/providers/bungie";

const handler = NextAuth({
	providers: [
		BungieProvider({
			clientId: process.env.BUNGIE_CLIENT_ID,
			clientSecret: process.env.BUNGIE_SECRET,
			headers: {
				"X-API-Key": process.env.BUNGIE_API_KEY,
			},
		}),
	],
});

export { handler as GET, handler as POST };
```

### How to reproduce

To reproduce this issue, follow these steps:

1. Create a nextjs app `npx create-next-app@latest`
2. Add the Bungie Provider configuration as shown in the code snippet above.
3. The error will appear in the route file if not run `npm run build`.

### Expected behavior

The Bungie Provider configuration should work without any errors. It should successfully authenticate users with the Bungie API using the provided credentials and headers. There should be no TypeScript error related to the headers property.

**Comment by mrxbox98** [comment]
I fixed the issue by doing
```ts
BungieProvider({
    clientId: process.env.BUNGIE_CLIENT_ID!,
    clientSecret: process.env.BUNGIE_SECRET!,
    httpOptions: {
        headers: {
            'X-API-Key': process.env.BUNGIE_API_KEY,
        },
    },
}),
```

But I'm still facing some issues with Bungie OAuth though I believe its unrelated to this.

**Comment by ThangHuuVu** [comment]
Sorry, we do have a documentation issue for Bungie. PR is welcome to fix the instruction for headers. 🙏
